packages/phone-number-privacy/common/src/interfaces/responses.ts

```diff
@@ -1,60 +1,99 @@
 import * as t from 'io-ts'
-import { Domain, DomainState } from '../domains'
 import {
   DisableDomainRequest,
   DomainQuotaStatusRequest,
   DomainRequest,
   DomainRestrictedSignatureRequest,
-} from './requests'
+  MatchmakingRequest,
+  OdisRequest,
+  PhoneNumberPrivacyRequest,
+  PnpQuotaRequest,
+  SignMessageRequest,
+} from '.'
+import { Domain, DomainState } from '../domains'
 
-export interface SignMessageResponse {
-  success: boolean
-  version?: string
-  signature?: string
+// Phone Number Privacy
+
+export interface SignMessageResponseSuccess {
+  success: true
+  version: string
   performedQueryCount?: number
   totalQuota?: number
   blockNumber?: number
+  signature?: string
 }
 
-export interface SignMessageResponseFailure extends SignMessageResponse {
+export interface SignMessageResponseFailure {
   success: false
+  version: string
   error: string
+  performedQueryCount?: number
+  totalQuota?: number
+  blockNumber?: number
+  signature?: string
 }
 
-export interface SignMessageResponseSuccess extends SignMessageResponse {
+export type SignMessageResponse = SignMessageResponseSuccess | SignMessageResponseFailure
+export interface GetQuotaResponseSuccess {
   success: true
+  version: string
+  performedQueryCount?: number
+  totalQuota?: number
+  blockNumber?: number
 }
 
-export interface GetQuotaResponse {
-  success: boolean
+export interface GetQuotaResponseFailure {
+  success: false
   version: string
-  performedQueryCount: number
-  totalQuota: number
+  error: string
 }
 
-export interface GetContactMatchesResponse {
-  success: boolean
-  matchedContacts: Array<{
-    phoneNumber: string
-  }>
+export type GetQuotaResponse = GetQuotaResponseSuccess | GetQuotaResponseFailure
+
+export interface GetContactMatchesResponseSuccess {
+  success: true
   version: string
+  matchedContacts: Array<{ phoneNumber: string }>
 }
+export interface GetContactMatchesResponseFailure {
+  success: false
+  version: string
+  error: string
+}
+
+export type GetContactMatchesResponse =
+  | GetContactMatchesResponseSuccess
+  | GetContactMatchesResponseFailure
+
+export type PhoneNumberPrivacyResponse<
+  R extends PhoneNumberPrivacyRequest = PhoneNumberPrivacyRequest
+> = R extends SignMessageRequest
+  ? SignMessageResponse
+  : never | R extends MatchmakingRequest
+  ? GetContactMatchesResponse
+  : never | R extends PnpQuotaRequest
+  ? GetQuotaResponse
+  : never
 
-export interface DomainRestrictedSignatureResponseSuccess {
+// Domains
+
+export interface DomainRestrictedSignatureResponseSuccess<D extends Domain = Domain> {
   success: true
   version: string
   signature: string
+  status: DomainState<D>
 }
 
-export interface DomainRestrictedSignatureResponseFailure {
+export interface DomainRestrictedSignatureResponseFailure<D extends Domain = Domain> {
   success: false
   version: string
   error: string
+  status: DomainState<D>
```

Comment:
What happens when the request fails to get the current quota status from the database? I'm guessing when will have this field be `undefined`. If so, this field should be `status?: DomainState<D>`